### PR TITLE
✨ feat: BPJS PRB resolved

### LINKED ISSUE #64: [TASK-ORDER] BPJS - PRB
## Background
BPJS - Rujuk balik integration

## Objective
Complete this endpoints
image

### PR DESCRIPTION
**Pull request by tukangk3tik**
Closes #64

### CODE REVIEW

File: server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import {ApiProperty} from "@nestjs/swagger";
+import {IsArray, IsEmail, IsNotEmpty, IsString, ValidateNested} from "class-validator";
+import {IObat, PrbContract} from "@core/3rdparty/bpjs/contract/prb.contract";
+import {Type} from "class-transformer";
+
+class PRBAddForm implements PrbContract {
```
Comment (theycantrevealus) [inline server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts:6]: Should we use this suffix? (Contract) for interfacing? Is it segmented naming or could be applied for many case? Should we need to define the project code pattern and rule?

**Review by theycantrevealus — COMMENTED** [review]
Good job. Please discuss some points. Thanks for your contribution

File: server/apps/core/src/modules/account/account.service.ts [hunk 2]
```diff
@@ -221,6 +221,7 @@ export class AccountService {
                   code: result.code,
                   first_name: result.first_name,
                   last_name: result.last_name,
+                  created_at: result.created_at,
```
[the remark this answers is not in the document]
Comment (tukangk3tik) [reply server/apps/core/src/modules/account/account.service.ts:224]: In PRB request, there's required field called `user` and should be numeric. In this project, the primary key is string and it contains alphanumeric characters. So, in our direct discussion, we agree to use `created_at` value (convert to unix epoch and must be unique) to fill the `user` field in the PRB request

File: server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts (hunk 1 above)
Comment (tukangk3tik) [reply server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts:6]: For now, i just use it for set to make sure the request and the schema have same required property when implement contract. In this case, mostly the field has same.

File: server/apps/core/src/modules/account/account.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (theycantrevealus) [reply server/apps/core/src/modules/account/account.service.ts:224]: But I think it should be under line 282. This line will affect to token creation only

File: server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts (hunk 1 above)
Comment (theycantrevealus) [reply server/apps/core/src/modules/3rdparty/bpjs/dto/prb/add.dto.ts:6]: I hope we can apply to all modules as a standard pattern

File: server/apps/core/src/modules/account/account.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tukangk3tik) [reply server/apps/core/src/modules/account/account.service.ts:224]: That's right. In our authentication middleware, it extract user data from token. So, i add `created_at` value to auth token, then before send PRB request, i convert `created_at` to unix epoch (integer) and using it as id (number only) as PRB API request expected.

Any security issue if i put the `created_at` in auth token?

**Review by theycantrevealus — COMMENTED** [review]
Ok

**Review by theycantrevealus — APPROVED** [review]

**Pull request merged**